lib/rack/request.rb

```diff
@@ -151,11 +157,11 @@ def request_method;  get_header(REQUEST_METHOD)                     end
       def query_string;    get_header(QUERY_STRING).to_s                  end
       def content_length;  get_header('CONTENT_LENGTH')                   end
```

Comment:
This is not the HTTP header it's an env key that must be supplied by the server per spec.

get_header is a misnomer, added in:

commit ef5546ce355f180c68bd0ce0d326b5e2c2d610ca
Author: Aaron Patterson <aaron.patterson@gmail.com>
Date:   Thu Aug 20 16:28:06 2015 -0700

It probably wants to be fixed to avoid this kind of confusion.